src/main/java/github/scarsz/discordsrv/api/commands/PluginCommandData.java

```diff
@@ -0,0 +1,81 @@
+package github.scarsz.discordsrv.api.commands;
+
+import lombok.Value;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import org.bukkit.plugin.Plugin;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * {@link CommandData} wrapper that includes the originating {@link Plugin}
+ */
+@Value
+public class PluginCommandData extends CommandData {
+
+    Plugin plugin;
+    CommandData commandData;
```

Comment:
The `CommandData` field isn't being used besides in `equals`, this is misleading as users might construct a CommandData object then pass it to PluginCommandData thinking it's details (eg. anything besides name and description) would be used, when that isn't the case.

Alternatively this class could be changed to not extend CommandData and use the field's CommandData instead.